src/server/controllers/user.js

```diff
@@ -9,32 +9,47 @@ const register = async (req, res) => {
     const { username, password } = req.body;
     const saltRounds = 10
     const hashedPassword = bcrypt.hashSync(password, saltRounds)
+    const foundUser = await prisma.user.findUnique({
+        where: {
+            username: username
+        }
+    })
+    if (foundUser) {
+        return res.status(200).json({ error: 'username already taken' })
+    }
+
     const createdUser = await prisma.user.create({
         data: {
             username: username,
             password: hashedPassword
         }
     })
-    res.json({ data: createdUser });
+
+    res.json({ created: createdUser });
 };
 
 const login = async (req, res) => {
+
     const { username, password } = req.body;
     const foundUser = await prisma.user.findUnique({
+        where: { username: username }
+    })
+    const userMovies = await prisma.movie.findMany({
         where: {
-            username:username
+            userId: foundUser.id
         }
     })
+    
     if (!foundUser) {
         return res.status(401).json({ error: 'Invalid username or password.' });
     }
     const passwordsMatch = bcrypt.compareSync(password, foundUser.password)
```

Comment:
Nice use of compareSync. I've just used compare but this seems like a more secure method that prevents timing attacks, which is handy.